DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky/src/cholesky_demo.cpp

```diff
@@ -0,0 +1,351 @@
+
+    // Create vectors to hold all the input and output matrices
+    std::vector<T> a_matrix;
+    std::vector<T> l_matrix;
+
+    a_matrix.resize(kAMatrixSize * kMatricesToDecompose);
+    l_matrix.resize(kLMatrixSize * kMatricesToDecompose);
+
+    std::cout << "Generating " << kMatricesToDecompose << " random ";
+    if constexpr (kComplex) {
+      std::cout << "complex ";
+    } else {
+      std::cout << "real ";
+    }
+    std::cout << "matri" << (kMatricesToDecompose > 1 ? "ces" : "x")
+              << " of size " << kRows << "x" << kColumns << " " << std::endl;
+
+    // Generate the random (hermitian and positive-definite) input matrices
+    srand(kRandomSeed);
+
+    for (int matrix_index = 0; matrix_index < kMatricesToDecompose;
+         matrix_index++) {
+      // Construct a single random hermitian and positive-definite matrix
+      // To do so we, we generate a hermitian matrix A where each element
+      // is between 0 and 1.
+      // Since A(i,j) < 1 by construction and a symmetric diagonally dominant
+      // matrix is symmetric positive definite we can be sure to have a
+      // symmetric diagonally dominant by adding nI to A
+      // A = A + n*eye(n);
+      // For complex matrices, the diagonal elements must be real.
+
+      // Random min and max values for the random floating-point value
+      // generation
+      constexpr float kRandomMin = 0;
+      constexpr float kRandomMax = 1;
+
+      int current_matrix = matrix_index * kAMatrixSize;
+
+      for (size_t row = 0; row < kRows; row++) {
+        for (size_t col = 0; col < kColumns; col++) {
+          float diag_scaling = (row == col) ? float{kRows} : 0;
+
+          int index = current_matrix + (col * kRows) + row;
+          int transpose_index = current_matrix + (row * kRows) + col;
+
+          if (col >= row) {
+            float random_real = RandomValueInInterval(kRandomMin, kRandomMax);
+#if COMPLEX == 0
+            a_matrix[index] = random_real + diag_scaling;
+#else
+            float random_imag =
+                row == col ? float{0}
+                           : RandomValueInInterval(kRandomMin, kRandomMax);
+            ac_complex<float> random_complex{random_real + diag_scaling,
+                                             random_imag};
+            a_matrix[index] = random_complex;
+#endif
+          } else {
+            // conjugate transpose
+#if COMPLEX == 0
+            a_matrix[index] = a_matrix[transpose_index];
+#else
+            a_matrix[index] = a_matrix[transpose_index].conj();
+#endif
+          }
+        }  // end of col
+      }    // end of row
+
+#ifdef DEBUG
+      std::cout << "A MATRIX " << matrix_index << std::endl;
+      for (size_t row = 0; row < kRows; row++) {
+        for (size_t col = 0; col < kColumns; col++) {
+          std::cout << a_matrix[current_matrix + (col * kRows) + row]
+                    << " ";
+        }  // end of col
+        std::cout << std::endl;
+      }  // end of row
+#endif
```

Comment:
Should this stay in?

Reply:
I like having these king of easy debugging info, even if anyone wants to do some modifications to the design or check that real data is being tested, it is just a matter of defining a DEBUG variable
But if you want it out, I can remove it

Reply:
Your choice! Just bringing attention to it.